File: modules/agent/core/src/main/java/dev/jvmguard/agent/config/transactions/FrequencyUnit.java
package dev.jvmguard.agent.config.transactions;

// do not rename enums
public enum FrequencyUnit {
    REQUESTS("Requests"),
    MINUTES("Minutes"),
    HOURS("Hours");

    private final String verbose;

    FrequencyUnit(String verbose) {
        this.verbose = verbose;
    }

    @Override
    public String toString() {
        return verbose;
    }
}

File: modules/backend/data/src/main/kotlin/dev/jvmguard/data/config/FrequencyUnit.kt
package dev.jvmguard.data.config

enum class FrequencyUnit(
    val description: String,
    val label: String,
    val multiplier: Int,
) {
    PER_SECOND("per second", " / s", 1),
    PER_MINUTE("per minute", " / m", 60),
    PER_HOUR("per hour", " / h", 60 * 60);

    override fun toString(): String = description
}
